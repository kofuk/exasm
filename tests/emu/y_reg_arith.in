lui r0, 0x1

lui r1, 0xde # Check mov
ori r1, 0xad
mov r2, r1
sw r2, (r0)
addi r0, 2

lui r1, 0x41 # Check not
ori r1, 0x10
not r1, r1
sw r1, (r0)
addi r0, 2

lui r1, 0xaa # Check xor
ori r1, 0xaa
lui r2, 0x92
ori r2, 0x49
xor r1, r2
sw r1, (r0)
addi r0, 2

lui r1, 0x30 # Check add
ori r1, 0x39
lui r2, 0x5b
ori r2, 0xa0
add r1, r2
sw r1, (r0)
addi r0, 2

lui r1, 0x5b # Check sub
ori r1, 0xa0
lui r2, 0x30
ori r2, 0x39
sub r1, r2
sw r1, (r0)
addi r0, 2

lui r1, 0xde # Check sl8
ori r1, 0xad
sl8 r2, r1
sw r2, (r0)
addi r0, 2

lui r1, 0xbe # Check sr8
ori r1, 0xef
sr8 r2, r1
sw r2, (r0)
addi r0, 2

lui r1, 0xde # Check sl
ori r1, 0xad
sl r2, r1
sw r2, (r0)
addi r0, 2

lui r1, 0xbe # Check sr
ori r1, 0xef
sr r2, r1
sw r2, (r0)
addi r0, 2

lui r1, 0xaa # Check and
ori r1, 0xaa
lui r2, 0x92
ori r2, 0x49
and r1, r2
sw r1, (r0)
addi r0, 2

lui r1, 0xaa # Check or
ori r1, 0xaa
lui r2, 0x92
ori r2, 0x49
or r1, r2
sw r1, (r0)
addi r0, 2
