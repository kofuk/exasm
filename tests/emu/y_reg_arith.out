0xde
0xad
0xbe
0xef
0x38
0xe3
0x8b
0xd9
0x2b
0x67
0xad
0x00
0x00
0xbe
0xbd
0x5a
0x5f
0x77
0x82
0x08
0xba
0xeb
